src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -408,6 +410,9 @@ private Binder VisitPropertyOrIndexerExpressionBody(BasePropertyDeclarationSynta
                         resultBinder = new InMethodBinder(accessor, resultBinder);
                     }
 
+                    resultBinder = resultBinder.WithUnsafeRegionIfNecessary(parent.Modifiers,
+                        isIteratorBody: accessor?.IsIterator == true);
```

Comment:
>accessor?.IsIterator == true [](http://example.com/codeflow?start=40&length=28)

I think an expression body is never an iterator. Consider asserting the fact in the ```if ((object)accessor != null)``` above and passing `false` here.